src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3499,18 +3540,24 @@ class CgroupUtils(object):
                                                 jobid)) as desc:
                         assigned[jobid][key]['limit_in_bytes'] = \
                             int(desc.readline())
+                    if assigned[jobid][key]['limit_in_bytes'] == mem_avail:
+                        assigned[jobid][key]['limit_in_bytes'] = 0
```

Comment:
Can you give the context of why this is being set to 0 if current value is equal to mem_avail?

Reply:
See comment above: if the per-job limit is that for the overarching "jobid" directory, then it was set because the job did not request memory and the limit was raised because default enforcement was disabled. Since the values reported as available to the server and scheduler are _lower_, it's impossible for an explicitly requested memory value to be this high.

These jobs did not request memory, so when determining how much memory was assigned, it is zero. Otherwise you'd have the paradoxical effect that if _one_ individual manages to get a job without an explicit memory request, then _no_ other job that asks even for one byte would be able to land on the node and get accepted by the cgroup hook. Since the _scheduler_ thinks that jobs that do not explicitly ask for memory ask for 0 bytes, that would lead to jobs being scheduled and rejected by MoM -- possibly several times, until the job was held.